xbmc/FileItem.cpp

```diff
@@ -124,6 +124,10 @@ CFileItem::CFileItem(const CEpgInfoTagPtr& tag)
   Initialize();
 
   m_bIsFolder = false;
+  if (tag->HasPVRChannel())
+  {
+    m_pvrChannelInfoTag = tag->ChannelTag();
```

Comment:
> If the channel is not set, lots of checks need to be changed

 can you point me to this checks, please?

> we are actually playing a channel here

then, a fileitem containing only a pvrchannel and no epg tag should be used (whatever used means, thats why I'm asking for the code places checks would need to be adapted)?

Maybe, instead of <code>g_application.PlayFile(CFileItem(m_progItem), "videoplayer");</code>, <code>g_application.PlayFile(CFileItem(m_progItem->ChannelTag()), "videoplayer");</code> is correct?